### PR TITLE
Add wrapper to allow injecting http response without using Robolectric

### PR DESCRIPTION
**Pull request by rpdome**
Inspired by (and took some code from) https://github.com/AzureAD/ad-accounts-for-android/pull/1719/files

This will allow us to directly inject http response to test cases.

### CODE REVIEW

**Comment by AdamBJohnsonx** [comment]
> Another alternative is to have the instantiated `InterceptedHttpClientWrapper` (a new one will be created for each PlatformComponent object) to take in MockHttpClient (or interceptors) directly.
> 
> We should just need one - having them for each endpoint is somewhat overkill imo.
> 
> Then we can kill the static part of the class.

Whichever, really.  From a test POV it doesn't really matter, though from an actual-usage POV there will be different configurations depending on the service you're talking to (timeouts, etc).

**Review by AdamBJohnsonx — APPROVED** [review]
I'm remarkable not picky here, now that I think about it.  Any way we choose to do it is fine, though I'd bias towards one that created fewer class files.

File: common4j/src/main/com/microsoft/identity/common/java/interfaces/IHttpClientWrapper.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+import lombok.NonNull;
+
+/**
+ * A wrapper interface around {@link HttpClient}, allowing us to inject responses in test cases.
+ */
+public interface IHttpClientWrapper {
+
+    @NonNull
+    HttpClient wrap(@NonNull final HttpClient client);
```
Comment (shahzaibj) [inline common4j/src/main/com/microsoft/identity/common/java/interfaces/IHttpClientWrapper.java:40]: probably javadoc here too

**Review by shahzaibj — APPROVED** [review]

**Review by shoatman — APPROVED** [review]
lgtm

**Review by iamgusain — APPROVED** [review]

**Pull request merged**